common/datasets/tedlium2/textual_data.py

```diff
@@ -0,0 +1,24 @@
+import os
+from functools import lru_cache
+from typing import Dict
+import pathlib
+
+from sisyphus import tk
+
+from .download import download_data_dict
+
+
+@lru_cache()
+def get_text_data_dict(output_prefix: str = "datasets") -> Dict[str, tk.Path]:
+    txt_dict = {}
+
+    lm_dir = download_data_dict(output_prefix=output_prefix)["lm_dir"]
+
+    txt_list = os.listdir(lm_dir)
+
+    for t in txt_list:
+        name = os.path.splitext(os.path.basename(t))[0]
+        path = tk.Path(os.path.abspath(t), cached=True)
+        txt_dict[name] = path
```

Comment:
from earlier up I remember
```
data_dict["lm_dir"] = download_tedlium2_job.out_lm_folder
```
i.e. `lm_dir` is a `Path` object which is the output of a download job. now you call `os.listdir` on the `Path` object.
Would this not fail when first starting the manager when the folder wouldn't be downloaded yet?
More severely:
```tk.Path(os.path.abspath(t))```
will hash the absolute path of the output of a sis job in the current graph. This constitutes a "hash bug", right?

As a remedy I would suggest (if possible) to explicitly list all expected filenames in code here and call 
```path = lm_dir.join_right(name)```

Reply:
Thanks for the quick fix!

To avoid hash-bugs and unintentional changes in the future, maybe should add this baseline to the hash test case.